infrahub_sdk/spec/object.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any, Optional
+
+from pydantic import BaseModel, Field
+
+from infrahub_sdk.client import InfrahubClient
+from infrahub_sdk.schema import MainSchemaTypes
+from infrahub_sdk.yaml import InfrahubFile, InfrahubFileKind
+
+
+class InfrahubObjectFileData(BaseModel):
+    kind: Optional[str] = None
+    data: list[dict[str, Any]] = Field(default_factory=list)
+
+    @classmethod
+    def enrich_node(cls, data: dict, context: dict) -> dict:
+        return data
+
+    @classmethod
+    async def create_node(
```

Comment:
In what way has this been used? I think it might be fine to create the menu but I'm wondering if we need to rethink the structure for other object types where the object we create is related to other objects. If those objects doesn't yet exist I think we'd need to create them first before creating the "base" object. Should be doable by looking at the relationships and which side if any is mandatory. Though it would also be nice if we could just feed everything to the backend as a "fix this in a single transaction" type of way. Not a blocker here and it could be that I don't completely understand how this gets used but I see some situations where it seems problematic now depending on how the relationships are defined.

Reply:
Yes I fully agree, this is not the final implementation
My end goal is to be able to create all these objects in memory and to have a engine that will know in which order they need to be created based on the schema.
This is very similar to what we need to build better generators.